docs/program-setup/serial/serial_communication.md

```diff
@@ -6,86 +6,100 @@ The Arm Mbed microcontroller on your board can communicate with a host PC over t
+The example prints "Hello World!" to the terminal after the reset.
 
-There are many terminal applications for Windows, including:
+To close the serial terminal, enter <kbd>Ctrl + C</kbd>.
 
-- [CoolTerm](http://freeware.the-meiers.org/) - this is the application we use in this example. We use it often because it usually "just works".
-- [Tera Term](http://sourceforge.jp/projects/ttssh2/files).
-- [PuTTY](http://www.chiark.greenend.org.uk/~sgtatham/putty/).
-- Some Windows PCs come with **Hyperterminal** installed.
+## Additional options of mbed sterm
 
-### Configuring the connection
+- If you have multiple boards connected:
+    1. Run `mbedls` to find the port of the board you want to use.
+    1. Run `mbed sterm -p <PORT>`.
 
-1. Plug in your Mbed board.
-1. Open CoolTerm.
-1. Click **Connect**. This opens up an 8-n-1 9600 baud connection to the first available serial port. If you have more than one board plugged in, you may need to change the port under **Options > Serial Port > Port**.
+- If your application uses a baud rate other than 9600, specify it with `-b <BAUDRATE>` when opening the serial terminal.
 
-Check your connection parameters:
+- To list all options, run `mbed sterm -h`:
 
-1. Select **Options > Serial Port**.
-1. You should see 9600 baud, 8 bits, 1 stop bit, no parity (9600-8-N-1).
-1. If you do not see your board, click **Re-Scan Peripherals**.
+    ```
+    usage: mbed sterm [-h] [-m TARGET] [-p PORT] [-b BAUDRATE] [-e ECHO] [-r] [-v] [-vv]
 
-Your terminal program is now configured and connected.
+    Open serial terminal to connected target (usually board), or connect to a user-specified COM port
 
-## Using terminal applications on Linux
+    optional arguments:
+    -h, --help            show this help message and exit
+    -m TARGET, --target TARGET
+                            Compile target MCU. Example: K64F, NUCLEO_F401RE,
+                            NRF51822...
+    -p PORT, --port PORT  Communication port. Default: auto-detect. Specifying
+                            this will also ignore the -m/--target option above.
+    -b BAUDRATE, --baudrate BAUDRATE
+                            Communication baudrate. Default: 9600
+    -e ECHO, --echo ECHO  Switch local echo on/off. Default: on
+    -r, --reset           Reset the targets (via SendBreak) before opening
+                            terminal.
+    -v, --verbose         Verbose diagnostic output
+    -vv, --very_verbose   Very verbose diagnostic output
+    ```
 
-CoolTerm should work under Linux. If for some reason it doesn't, you can try one of the following:
+## Auto-opening a serial terminal after compilation
 
-- [Minicom](https://help.ubuntu.com/community/Minicom).
-- [GNU Screen](https://www.gnu.org/software/screen/manual/screen.html).
+To compile, flash and open a serial terminal in one command line:
 
-## Minimal Printf
+```
+mbed compile -t <TOOLCHAIN> -m <TARGET> --flash --sterm
```

Comment:
You could also show usage of `-m detect`

Reply:
It's a nice tip. But `-m detect` seems like an undocumented feature (not shown in `mbed compile --help`), so I'm not sure if we should suggest it.